### PR TITLE
fix: adjust test snapshots and small refactors

### PR DESCRIPTION
**Pull request by drbh**
This PR updates the test snapshots and to fix tests failing in CI

### CODE REVIEW

File: server/Makefile-selective-scan [hunk 1]
```diff
@@ -26,3 +26,6 @@ install-selective-scan: install-causal-conv1d build-selective-scan
 	cd mamba && pip install .
 
 build-all: build-causal-conv1d build-selective-scan
+
+install-ssm: install-causal-conv1d install-selective-scan
+	@echo "Selective scan model installed"
```
Comment (Narsil) [inline server/Makefile-selective-scan:31]: No needed here.

File: server/text_generation_server/models/causal_lm.py [hunk 2]
```diff
@@ -609,15 +609,16 @@ def fallback(
             truncation_side="left",
             trust_remote_code=trust_remote_code,
         )
+        device_map = (
+            "auto"
+            if torch.cuda.is_available() and torch.cuda.device_count() > 1
+            else None
+        )
```
Comment (Narsil) [inline server/text_generation_server/models/causal_lm.py:616]: To revert.

File: server/text_generation_server/models/causal_lm.py [hunk 3]
```diff
@@ -628,6 +629,11 @@ def fallback(
         ):
             model = model.cuda()
 
+        # if device_map is "auto", it's unclear which device the model is on
+        # therefore, we need to get the device the model is on after loading
+        if device_map is not None:
+            device = next(model.parameters()).device
+
```
Comment (Narsil) [inline server/text_generation_server/models/causal_lm.py:636]: To revert.

File: server/text_generation_server/models/seq2seq_lm.py [hunk 4]
```diff
@@ -639,21 +639,28 @@ def fallback(
...
             trust_remote_code=trust_remote_code,
         )
         if torch.cuda.is_available() and torch.cuda.device_count() == 1:
             model = model.cuda()
 
+        # if device_map is "auto", it's unclear which device the model is on
+        # therefore, we need to get the device the model is on after loading
+        if device_map is not None:
+            device = next(model.parameters()).device
+
```
Comment (Narsil) [inline server/text_generation_server/models/seq2seq_lm.py:663]: To revert.

File: server/text_generation_server/models/causal_lm.py (hunk 2 above)
Comment (drbh) [reply server/text_generation_server/models/causal_lm.py]: reverted

File: server/text_generation_server/models/causal_lm.py (hunk 3 above)
Comment (drbh) [reply server/text_generation_server/models/causal_lm.py]: reverted

File: server/text_generation_server/models/seq2seq_lm.py (hunk 4 above)
Comment (drbh) [reply server/text_generation_server/models/seq2seq_lm.py]: reverted

File: server/Makefile-selective-scan (hunk 1 above)
Comment (drbh) [reply server/Makefile-selective-scan]: reverted

**Pull request merged**